Lib/ctypes/util.py

```diff
@@ -306,10 +306,42 @@ def _findLib_ld(name):
                 pass  # result will be None
             return result
 
+        def _findLib_walkldpath(name):
+            # See issue #21622
+            def _is_elf(filepath):
+                try:
+                    with open(filepath, 'rb') as fh:
+                        return fh.read(4) == b'\x7fELF'
+                except:
+                    return False
+
+            from glob import glob
+            if os.path.isabs(name):
+                return name
+            if name.startswith("lib") and name.endswith(".so"):
+                suffixes = [ '.[0-9]*' ]
+                libname = name
+            else:
+                suffixes = ['.so', '.so.[0-9]*', '.musl-*.so.[0-9]*']
+                libname = 'lib'+name
+            # search LD_LIBRARY_PATH list or default locations
+            paths = os.environ.get('LD_LIBRARY_PATH', '').split(':') + ['/lib', '/usr/local/lib', '/usr/lib']
```

Comment:
Those are not the default locations. The default locations can change but they usually are `/lib` and `/usr/lib`. There is also ` /etc/ld-musl-$ARCH.path`, and for GNU, `/etc/ld.so.conf`, which you are ignoring, and should definitely be honored.

Reply:
This is only meant as fallback for musl libc, a last resort when no other ways work, and for musl, the [default locations]( https://git.musl-libc.org/cgit/musl/tree/ldso/dynlink.c?id=a5aff1972c9e3981566414b09a28e331ccd2be5d#n1074) are:
```c
if (!sys_path) sys_path = "/lib:/usr/local/lib:/usr/lib";
```

For GNU and `/etc/ld.so.conf` case there should be a working `ldconfig -p` and will be covered by `_findSoname_ldconfig`.

However, the `/etc/ld-musl-$ARCH.path` should probably be honored, even I never seen it used in any real life situation.

Reply:
`/etc/ld-musl-$ARCH.path` is now honored